### PR TITLE
Add Cloud compatibility notes to guides

### PR DESCRIPTION
**Pull request by ptgott**
Add Details boxes to some guides that do not mention Teleport
Cloud, indicating how each guide relates to Teleport Cloud users.

### CODE REVIEW

File: docs/pages/application-access/guides/api-access.mdx [hunk 1]
```diff
@@ -8,6 +8,10 @@ description: How to access REST APIs with Teleport Application Access.
 Teleport Application Access can be used to access applications' (REST or Teleport's own gRPC) APIs with
 tools like [curl](https://man7.org/linux/man-pages/man1/curl.1.html) or Postman.
 
+<Details scopeOnly={true} scope={["cloud"]} title="Teleport Cloud Compatibility">
```
Comment (zmb3) [inline docs/pages/application-access/guides/api-access.mdx:11]: Is the idea that we're going to put a note like this on every guide that is compatible with cloud?

1-2 years from now I will expect that 90% or more of our guides work on cloud, won't this get repetitive? Would it be easier to assume cloud compatibility and only call out cases that don't work on cloud?

File: docs/pages/getting-started/linux-server.mdx [hunk 2]
```diff
@@ -206,15 +214,15 @@ $ tsh ls
...
+### Add a Node to the cluster
 
 Generate a short-lived dynamic join token using `tctl`:
 
 ```code
 $ tctl tokens add --type=node
 ```
 
-Bootstrap a new node:
+Bootstrap a new Node:
```
Comment (ulysseskan) [inline docs/pages/getting-started/linux-server.mdx:225]: I probably wouldn't capitalize node unless it were part of a title.

File: docs/pages/getting-started/linux-server.mdx [hunk 3]
```diff
@@ -206,15 +214,15 @@ $ tsh ls
 $ tsh ssh root@node-name
 ```
 
-### Add a node to the cluster
+### Add a Node to the cluster
```
Comment (ulysseskan) [inline docs/pages/getting-started/linux-server.mdx:217]: If you're going to capitalize node in this title, you might want to capitalize cluster as well.

Comment (ptgott) [reply docs/pages/getting-started/linux-server.mdx:217]: See my note re: "Node" above

**Comment by ptgott** [comment]
@zmb3 @ulysseskan Thanks for reviewing—this should be ready for another look.

**Review by zmb3 — APPROVED** [review]

**Review by ulysseskan — APPROVED** [review]
Looks good to me, Paul.  Thanks!

**Comment by russjones** [comment]
@ptgott Looks like this is approved, can we merge?

**Pull request merged**